### PR TITLE
Create tests for communication between server and gui

### PR DESCRIPTION
**Pull request by Njord201**
Hello mates ! In this pull request i've added many tests for the communication between the server and the gui.

You can see the result:

Lines:
image
Branches:
image

### CODE REVIEW

File: tests/server/tests/gui/commands/tests_bct.c [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+** File description:
+** test_bct
+*/
+
+#include <criterion/criterion.h>
+#include <criterion/redirect.h>
+#include "types.h"
+#include "app/app.h"
+#include "gui/gui.h"
+#include "gui/communication.h"
```
Comment (mathieurobert1) [inline tests/server/tests/gui/commands/tests_bct.c:13]:
```suggestion
#include <criterion/criterion.h>
#include <criterion/redirect.h>

#include "types.h"
#include "app/app.h"
#include "gui/gui.h"
#include "gui/communication.h"
```

File: tests/server/tests/gui/commands/tests_mct.c [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+** File description:
+** test_mct
+*/
+
+#include <criterion/criterion.h>
+#include <criterion/redirect.h>
+#include "types.h"
+#include "app/app.h"
+#include "gui/gui.h"
+#include "gui/communication.h"
```
Comment (mathieurobert1) [inline tests/server/tests/gui/commands/tests_mct.c:13]:
```suggestion
#include <criterion/criterion.h>
#include <criterion/redirect.h>

#include "types.h"
#include "app/app.h"
#include "gui/gui.h"
#include "gui/communication.h"
```

File: tests/server/tests/gui/commands/tests_sgt.c [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/*
+** EPITECH PROJECT, 2024
+** Zappy
+** File description:
+** test_sgt
+*/
+
+#include <criterion/criterion.h>
+#include <criterion/redirect.h>
+#include "types.h"
```
Comment (AxelF44) [inline tests/server/tests/gui/commands/tests_sgt.c:10]:
```suggestion
#include <criterion/criterion.h>
#include <criterion/redirect.h>

#include "types.h"
```

File: tests/server/tests/gui/commands/tests_sst.c [hunk 4]
```diff
@@ -0,0 +1,89 @@
+/*
+** EPITECH PROJECT, 2024
+** Zappy
+** File description:
+** test_sst
+*/
+
+#include <criterion/criterion.h>
+#include <criterion/redirect.h>
+#include "types.h"
```
Comment (AxelF44) [inline tests/server/tests/gui/commands/tests_sst.c:10]:
```suggestion
#include <criterion/criterion.h>
#include <criterion/redirect.h>

#include "types.h"
```

**Review by AxelF44 — CHANGES_REQUESTED** [review]

File: tests/server/tests/gui/commands/tests_sgt.c [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+#include "app/app.h"
+#include "gui/gui.h"
+#include "gui/communication.h"
```
Comment (mathieurobert1) [inline tests/server/tests/gui/commands/tests_sgt.c:13]:
```suggestion
#include <criterion/criterion.h>
#include <criterion/redirect.h>

#include "types.h"
#include "app/app.h"
#include "gui/gui.h"
#include "gui/communication.h"
```

File: tests/server/tests/gui/commands/tests_sst.c [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+#include "app/app.h"
+#include "gui/gui.h"
+#include "gui/communication.h"
```
Comment (mathieurobert1) [inline tests/server/tests/gui/commands/tests_sst.c:13]:
```suggestion
#include <criterion/criterion.h>
#include <criterion/redirect.h>

#include "types.h"
#include "app/app.h"
#include "gui/gui.h"
#include "gui/communication.h"
```

**Review by mathieurobert1 — CHANGES_REQUESTED** [review]
Bad include norm

**Review by mathieurobert1 — APPROVED** [review]

**Review by Thomaltarix — APPROVED** [review]

**Review by AxelF44 — CHANGES_REQUESTED** [review]
I accept that for a lollipop with orange!

**Review by AxelF44 — APPROVED** [review]

**Pull request merged**